src/exec/contract/ClearingHouse.ts

```
@@ -0,0 +1,144 @@
+import ClearingHouseArtifact from "@perp/contract/build/contracts/src/ClearingHouse.sol/ClearingHouse.json"
+
+import { ClearingHouse } from "../../type"
+import { getContract } from "../../util/contract"
+import { Metadata } from "../../util/metadata"
+
+import { Signer } from "@ethersproject/abstract-signer"
+import { TransactionReceipt } from "@ethersproject/abstract-provider"
+import { BigNumber, utils } from "ethers"
+import { formatError, formatProperty } from "../../util/format"
+import { EmptyArg } from "../../util/functionsMap"
+import { getSuggestedGas } from "../../util/provider"
+
+export class OpenPositionArg implements EmptyArg {
+    constructor(
+        readonly amm: string,
+        readonly side: number,
+        readonly quoteAssetAmount: string,
+        readonly leverage: string,
+        readonly baseAssetAmountLimit: string,
+    ) {}
+
+    verify() {
+        if (
+            this.amm === undefined ||
+            this.side === undefined ||
+            this.leverage === undefined ||
+            this.baseAssetAmountLimit === undefined ||
+            this.quoteAssetAmount === undefined
+        ) {
+            console.log(formatError("args for openPosition are incomplete"))
+            throw ""
```

let's throw an error with string here such as `throw new Error(<REASON>)`